extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/ColumnChunkPageStore.java

```diff
@@ -171,17 +172,52 @@ public boolean usesDictionaryOnEveryPage() {
     @Override
     public void close() {}
 
-    FillContext innerFillContext(@NotNull final FillContext context) {
-        // TODO(deephaven-core#4836): Call this method from the appropriate place in the implementation of
-        // getPageContaining to populate the context object
-        return ((PagingContextHolder) context)
-                .updateInnerContext(this::fillContextUpdater);
+    /**
+     * Take an object of {@link PagingContextHolder} and populate the inner context with values from
+     * {@link #columnChunkReader}
+     */
+    final void innerFillContext(@Nullable final FillContext context) {
+        if (context != null) {
+            ((PagingContextHolder) context).updateInnerContext(this::fillContextUpdater);
+        }
+    }
+
+    /**
+     * Return true if this class allocated the current inner context and therefore the context can be reused for reading
+     * pages.
+     */
+    private static boolean isFillContextCompatible(@Nullable final Context currentInnerContext) {
+        return currentInnerContext instanceof ChannelContextWrapper;
+    }
+
+    static SeekableChannelsProvider.ChannelContext getChannelContext(@Nullable final FillContext fillContext) {
+        if (fillContext instanceof PagingContextHolder) {
```

Comment:
Why would this be `false` unless `fillContext == null`?

Reply:
I am doing the cast in next statement`(PagingContextHolder) fillContext`, so I thought the `instanceOf` check will be safer. But I can change it to null because it might be an error if we don't get a PagingContextHolder object here. So better not to hide this in an if check.